### PR TITLE
remove self hosted etcd functionality

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: cmd/bootkube/recover.go [hunk 1]
```diff
@@ -13,7 +13,6 @@ import (
 
 	"github.com/kubernetes-incubator/bootkube/pkg/bootkube"
 	"github.com/kubernetes-incubator/bootkube/pkg/recovery"
-	"github.com/kubernetes-incubator/bootkube/pkg/util/etcdutil"
```
Comment (ericchiang) [inline cmd/bootkube/recover.go]: Is etcdutil used at all anymore?

Comment (ericchiang) [reply cmd/bootkube/recover.go]: Can we remove it too?

Comment (rphillips) [reply cmd/bootkube/recover.go]: good catch. It is not. I'll remove the code.

Comment (rphillips) [reply cmd/bootkube/recover.go]: fixed in 3c946bc

**Comment by ericchiang** [comment]
Thanks. Can you squash?

**Comment by rphillips** [comment]
Yes... I'll squash right after the tests get done.

**Comment by rphillips** [comment]
@ericchiang removed the unused vendor modules in a separate commit

**Comment by rphillips** [comment]
made a trivial tweak to the roadmap docs to remove self hosted etcd

**Comment by rphillips** [comment]
coreosbot run e2e

**Comment by xiang90** [comment]
@rphillips 

is `hack/multi-node/bootkube-test-recovery-self-hosted-etcd` also removed?

**Comment by rphillips** [comment]
another good find... I removed that file and squashed it into the first commit.

**Comment by rphillips** [comment]
coreosbot run e2e

**Comment by ericchiang** [comment]
Hmm, API server never came back up

```
[148350.353894] golang[6]: === RUN   TestReboot
[148452.477483] golang[6]: --- PASS: TestReboot (102.12s)
[148452.478063] golang[6]: 	reboot_test.go:22: rebooting 2 nodes
[148452.478307] golang[6]: === RUN   TestSmoke
[148522.907035] golang[6]: E0112 20:09:13.912370    2124 nginx_workload.go:84] Error getting deployment nginx-1txrh: Get https://54.89.231.177:6443/apis/extensions/v1beta1/namespaces/bootkube-e2e-7d7e869a/deployments/nginx-1txrh: dial tcp 54.89.231.177:6443: getsockopt: connection refused
[148532.906674] golang[6]: E0112 20:09:23.912044    2124 nginx_workload.go:84] Error getting deployment nginx-1txrh: Get https://54.89.231.177:6443/apis/extensions/v1beta1/namespaces/bootkube-e2e-7d7e869a/deployments/nginx-1txrh: dial tcp 54.89.231.177:6443: getsockopt: connection refused
[148542.906786] golang[6]: E0112 20:09:33.912171    2124 nginx_workload.go:84] Error getting deployment nginx-1txrh: Get https://54.89.231.177:6443/apis/extensions/v1beta1/namespaces/bootkube-e2e-7d7e869a/deployments/nginx-1txrh: dial tcp 54.89.231.177:6443: getsockopt: connection refused
```

**Comment by rphillips** [comment]
hmm. the build last night was green... I wonder if we should add this to the flaky issue.

**Comment by rphillips** [comment]
coreosbot run e2e

**Comment by xiang90** [comment]
> hmm. the build last night was green... I wonder if we should add this to the flaky issue.

let us add it.

**Comment by rphillips** [comment]
added to https://github.com/kubernetes-incubator/bootkube/issues/824

**Comment by rphillips** [comment]
Same flake that Diego reported as well.

**Comment by klausenbusk** [comment]
It seems like a few things was forgotten in this PR, I have opened a new PR to fix that: https://github.com/kubernetes-incubator/bootkube/pull/852

**Comment by carlos-licea** [comment]
Naïve question, why was this removed? I've been requested to have a similar setup. I have to wonder, what's the case for on-host etcd rather than self-contained?

**Pull request merged**